src/os/ui/draw/draw.js

```diff
@@ -1,8 +1,143 @@
 goog.provide('os.ui.draw');
 
+goog.require('ol.Feature');
+goog.require('ol.geom.Polygon');
+goog.require('ol.style.Fill');
+goog.require('ol.style.Stroke');
+goog.require('ol.style.Style');
+goog.require('os.geo.jsts.OLParser');
+goog.require('os.style.area');
+
 
 /**
  * The menu to display when drawing interaction completes.
  * @type {os.ui.menu.Menu|undefined}
  */
 os.ui.draw.MENU = undefined;
+
+
+/**
+ * Build a list of features to represent the search grid that intersects with the drawn geometry
+ *
+ * @param {ol.Feature} feature The source feature to trace.
+ * @param {osx.ui.draw.GridOptions} options Config object for grid generation
+ * @return {Array.<ol.Feature>}
+ *
+ * @suppress {accessControls} To allow direct access to feature metadata.
+ */
+os.ui.draw.getGridFromFeature = function(feature, options) {
+  var valid = function(options) {
+    var b = (typeof options != 'undefined' && options != null);
+    if (b) b = (typeof options.detail == 'number' && options.detail > 0.0); // enforce numeric, positive, non-null, non-zero;
+    if (b) b = (typeof options.max == 'number' && options.max > 0.0); // enforce numeric, positive, non-null, non-zero;
+    if (b) b = (typeof options.style == 'object');
+    return b; // any failure skips next tests
+  };
+
+  if (!feature || !options || !valid(options)) return null;
+
+  var detail = options.detail;
+  var max = options.max;
+
+  var features = [];
+  var geo = feature.getGeometry();
+
+  geo.toLonLat(); // convert to lon/lat so grid 'detail' (in degrees) can be applied easily
+  var extent = geo.getExtent(); // build a simplified box around the search geometry
+  geo.osTransform(); // undo transformation so copyFeature() succeeds
+
+  // snap box coordinates to the nearest "detail" degrees
+  var snap = function(l, d, ceil) {
+    if ((l % d) != 0.0) return Math.floor(l / d) * d + (ceil ? d : 0.0);
+    return l;
+  };
+
+  extent[0] = snap(extent[0], detail, false); // lon min
+  extent[1] = snap(extent[1], detail, false); // lat min
+  extent[2] = snap(extent[2], detail, true); // lon max
+  extent[3] = snap(extent[3], detail, true); // lat max
+
+  // don't continue building the grid if it would create too many boxes
+  if ((Math.ceil(Math.abs(extent[2] - extent[0]) / detail) *
+      Math.ceil(Math.abs(extent[3] - extent[1]) / detail)) > max) return null;
+
+  var cfg = {
+    lon: {
+      n: (extent[0] < extent[2]) ? extent[0] : extent[2],
+      x: (extent[0] < extent[2]) ? extent[2] : extent[0]
+    },
+    lat: {
+      n: (extent[1] < extent[3]) ? extent[1] : extent[3],
+      x: (extent[1] < extent[3]) ? extent[3] : extent[1]
+    }
+  }; // sort the data so the loops later are faster
+
+  var prop = {};
+  goog.object.extend(prop, feature.values_);
```

Comment:
To future-proof this, you might want to use `var prop = ol.obj.assign({}, feature.values_);`

Reply:
I wanted the projection, altitude mode (clamp to ground, etc) and - more importantly - any settings that I didn't know about.